### PR TITLE
WIXFEAT:3933 - Make WixBundleManufacturer variable writable

### PR DESCRIPTION
**Pull request by rseanhall**

### CODE REVIEW

File: src/burn/engine/registration.cpp [hunk 1]
```diff
@@ -616,112 +622,110 @@ extern "C" HRESULT RegistrationSessionBegin(
...
         }
 
         // Publisher: provided by UI
-        if (pRegistration->sczPublisher)
-        {
-            hr = RegWriteString(hkRegistration, REGISTRY_BUNDLE_PUBLISHER, pRegistration->sczPublisher);
-            ExitOnFailure1(hr, "Failed to write %ls value.", REGISTRY_BUNDLE_PUBLISHER);
-        }
+        hr = GetBundleManufacturer(pRegistration, pVariables, &sczPublisher);
+        hr = RegWriteString(hkRegistration, REGISTRY_BUNDLE_PUBLISHER, SUCCEEDED(hr) ? sczPublisher : pRegistration->sczPublisher);
```
Comment (barnson) [inline src/burn/engine/registration.cpp]: Any reason to skip normal ExitOnFailure handling? If GetBundleManufacturer, it's probably catastrophic anyway...

Comment (rseanhall) [inline src/burn/engine/registration.cpp]: Just handling it like GetBundleName above.

Comment (barnson) [inline src/burn/engine/registration.cpp]: If only I had post-commit telepathy to know why that was done that way...

Comment (rseanhall) [inline src/burn/engine/registration.cpp]: What are the chances that @robmen remembers? http://wix.codeplex.com/SourceControl/changeset/d08bc17758b3abd0fd9c49cfe998dcb02daaf6d7

Comment (robmen) [inline src/burn/engine/registration.cpp]: Avoid non fatal error, I think.

**Pull request merged**